lib/index.js

```diff
@@ -122,7 +123,12 @@ class Linter {
     };
 
     try {
-      preprocess(stripBom(options.source), {
+      const source = stripBom(options.source);
+      const preprocessStr =
+        options.moduleId.endsWith('.js') || options.moduleId.endsWith('.ts')
+          ? searchAndExtractHbs(source)
+          : source;
```

Comment:
Perhaps we should target this at `ember-template-recast` itself since that lib wants to be able to update inline templates and we are actively working on updating here to use `ember-template-recast`, see #873

Reply:
I have tried `ember-template-recast`, correct me if i'm wrong but this will parse and transform **hbs** source to **hbs** source, here the input isn't **hbs**, we need to convert the js/ts content to hbs first !

Reply:
you need to extract template text using (for example - ember-meta-explorer), and after - use it as souce for template-lint logic, including ember-template-recast

Reply:
@lifeart I have tried `ember-meta-explorer` and it seems just a **wrapper** of `parser`(babylon..., to AST) + `traverser`(babel-traverser with some pre-defined params, plugins ...) + `generator`(babel-generator, from AST to the original code format).

I have scanned all the test examples, i can't find any thing that go from `js/ts` to `hbs` :(

First, we need to figure out how to deal with **Call Expression** formats (e.g. `hbs('template content')`), please take a look at https://github.com/ember-template-lint/ember-template-lint/pull/888#discussion_r342510569

-> i'm waiting for Robert response, if we will not support **Call Expression** formats, i don't think that we have to use a parser to **just extract tagged literal templates** !

Reply:
flashios09

```js
const traverse = require("@babel/traverse");
const { parseScriptFile } = require("ember-meta-explorer");

const fileContent = 'hbs`hello`';
const templates = [];
const ast = parseScriptFile(fileContent);

traverse(ast, {
  TaggedTemplateExpression(path) {
    const node = path.node;
    if (node.tag && node.tag.type === 'Identifier' && node.tag.name === 'hbs') {
      const location = node.quasi.quasis[0].value.loc;
      const templateText = node.quasi.quasis[0].value.raw;
      templates.push([templateText , location]);
    }
  }
});
 
 
console.log(templates);
```